### PR TITLE
Include rlsr json

### PR DESCRIPTION
**Pull request by matthias-reis**
See https://jira.xing.hh/browse/WEB-1136

### CODE REVIEW

File: new/analyse/add-dependencies.ts [hunk 1]
```diff
@@ -1,64 +1,73 @@
...
-        [packageName]: currentPackage,
-      };
-    },
-    env.packages
-  );
+    return {
+      ...accumulator,
+      [packageName]: currentPackage,
+    };
+  }, env.packages);
```
Comment (santiph) [inline new/analyse/add-dependencies.ts:70]: this should be ramda-cloned to keep this a pure function

File: new/analyse/add-dependencies.ts [hunk 2]
```diff
@@ -1,64 +1,73 @@
...
-      currentPackage.dependsOn.push(
-        ...Object.entries(currentPackage.packageJson.dependencies || {})
-          .filter(([name]) => packagesNames.includes(name))
-          .map(([name, range]) => ({
-            type: 'default' as RelatedPackageTypes,
-            name,
-            range,
-          }))
-      );
+  const packages = packagesNames.reduce((accumulator, packageName) => {
```
Comment (santiph) [inline new/analyse/add-dependencies.ts:23]: I am switching to `es2019`, so fromEntries() could be used here, if it improves readability

File: new/analyse/add-dependencies.ts [hunk 3]
```diff
@@ -1,64 +1,73 @@
...
-      currentPackage.dependsOn.push(
-        ...Object.entries(currentPackage.packageJson.devDependencies || {})
-          .filter(([name]) => packagesNames.includes(name))
-          .map(([name, range]) => ({
-            type: 'dev' as RelatedPackageTypes,
-            name,
-            range,
-          }))
-      );
+    const dependencies: RelatedPackageDependsOn[] = [
```
Comment (santiph) [inline new/analyse/add-dependencies.ts:26]: You can declare this as `const dependencies: Package['dependsOn']`, so you don't have to import `RelatedPackageDependsOn`

File: new/analyse/add-dependencies.ts [hunk 4]
```diff
@@ -1,64 +1,73 @@
...
+      // all dependencies
+      ...Object.entries(currentPackage.packageJson.dependencies || {}).map(
+        ([name, range]) => ({
```
Comment (santiph) [inline new/analyse/add-dependencies.ts:29]: Are external dependencies also needed/wanted here?

File: new/analyse/spec/add-dependencies.spec.ts [hunk 5]
```diff
@@ -93,4 +88,122 @@ describe('addDependencies Module', () => {
       expect(result).toEqual(expectedEnv);
     });
   });
+
+  describe('with rlsr.json data', () => {
```
Comment (santiph) [inline new/analyse/spec/add-dependencies.spec.ts:92]: can you add a `without rlsr.json data` section as well?

File: new/analyse/spec/add-dependencies.spec.ts [hunk 6]
```diff
@@ -93,4 +88,122 @@ describe('addDependencies Module', () => {
...
+    });
+    it('has a result', () => {
+      expect(result).toBeDefined();
+    });
+    it('takes the dependency from rlsr.json if it is `*`', () => {
+      const found = dependencies.find(
+        (dependency) => dependency.name === 'mockPreviouslyAvailable'
+      );
+      expect(found).toBeDefined();
+      expect(found?.range).toEqual('1 - 2');
```
Comment (santiph) [inline new/analyse/spec/add-dependencies.spec.ts:165]: You could use `expect.objectContaining(object)` as well (https://jestjs.io/docs/expect#expectobjectcontainingobject) to test in one assertion multiple properties 

```javascript
expect(result?.packages?.mockPackageUnderTest?.dependsOn).toEqual(expect.objectContaining({
  name: 'mockPreviouslyAvailable'
  range: '1 - 2'
}))
```

File: new/analyse/spec/add-dependencies.spec.ts [hunk 7]
```diff
@@ -93,4 +88,122 @@ describe('addDependencies Module', () => {
...
+      expect(found?.range).toEqual('*');
+    });
+  });
 });
+
+const createMockPackage = (
+  name: string,
+  dependencies: Record<string, string> = {},
+  peerDependencies: Record<string, string> = {},
+  override: Partial<Package> = {}
```
Comment (santiph) [inline new/analyse/spec/add-dependencies.spec.ts:196]: 👌

File: new/analyse/add-dependencies.ts (hunk 3 above)
Comment (matthias-reis) [reply new/analyse/add-dependencies.ts:26]: I like the explicit notation much better. Especially, you directly jump there when you click on it.

File: new/analyse/add-dependencies.ts (hunk 2 above)
Comment (matthias-reis) [reply new/analyse/add-dependencies.ts:26]: Didn't touch that. just removed the two last attributes. So I'd leave that here.

File: new/analyse/add-dependencies.ts (hunk 4 above)
Comment (matthias-reis) [reply new/analyse/add-dependencies.ts:32]: They're filtered out below for both types at the same time.

File: new/analyse/add-dependencies.ts (hunk 1 above)
Comment (matthias-reis) [reply new/analyse/add-dependencies.ts]: That line was already there. It just got rearranged by prettier. But I added the `clone` lines.

File: new/analyse/spec/add-dependencies.spec.ts (hunk 5 above)
Comment (matthias-reis) [reply new/analyse/spec/add-dependencies.spec.ts:92]: That's already covered by your tests above, isn't it?

File: new/analyse/spec/add-dependencies.spec.ts (hunk 6 above)
Comment (matthias-reis) [reply new/analyse/spec/add-dependencies.spec.ts:165]: But it's an array. Doesn't seem to work according to their specs. Also `arrayContaining` doesn't work.

**Review by matthias-reis — COMMENTED** [review]
/

File: new/analyse/spec/add-dependencies.spec.ts (hunk 5 above)
Comment (santiph) [reply new/analyse/spec/add-dependencies.spec.ts:92]: I think they are. I think it'll be better to have them wrapped in a `describe('without rlsr.json data', () => { ... })`

Comment (matthias-reis) [reply new/analyse/spec/add-dependencies.spec.ts:93]: Cool. Changed that line.

**Review by santiph — APPROVED** [review]

**Pull request merged**